cpp/open3d/t/geometry/PointCloud.cpp

```diff
@@ -314,6 +314,44 @@ PointCloud PointCloud::VoxelDownSample(
     return pcd_down;
 }
 
+PointCloud PointCloud::UniformDownSample(size_t every_k_points) const {
+    if (every_k_points == 0) {
+        utility::LogError(
+                "Illegal sample rate, every_k_points must be larger than 0.");
+    }
+
+    const int64_t length = GetPointPositions().GetLength();
+
+    PointCloud pcd_down(GetDevice());
+    for (auto &kv : GetPointAttr()) {
+        pcd_down.SetPointAttr(
+                kv.first,
+                kv.second.Slice(0, 0, length, (int64_t)every_k_points));
+    }
+
+    return pcd_down;
+}
+
+PointCloud PointCloud::RandomDownSample(double sampling_ratio) const {
```

Comment:
1. For reproducible experiments, we need the ability to set the random seed for all RNG calls by Open3D. One way to achieve this is with a singleton RNG, with some care for concurrent access. 
2. CUDA has an RNG that will likely be faster for CUDA tensors.